security/kerberos_pac.ksy

```diff
@@ -0,0 +1,542 @@
+      - id: home_directory_drive_body
+        type: unicode_string_body
+      - id: group_ids
+        -orig-id: GroupIds
+        type: rpc_group_membership
+        if: group_count > 0
+      - id: logon_server_body
+        type: unicode_string_body
+      - id: logon_domain_name_body
+        type: unicode_string_body
+      - id: logon_domain_id
+        type: sid
+      - id: extra_sids
+        type: rpc_extra_sids
+        if: (sid_count > 0) and (user_flags & netlogon_extra_sids_user_flag != 0)
+      - id: resource_group_domain_sid
+        -orig-id: ResourceGroupDomainSid
+        type: sid
+        if: user_flags & netlogon_resource_groups_user_flag != 0
+      - id: resource_group_ids
+        -orig-id: ResourceGroupIds
+        type: rpc_resource_group_ids
+        if: (user_flags & netlogon_resource_groups_user_flag != 0) and (resource_group_count > 0)
+  pac_client_info:
+    doc: "See [MS-PAC] 2.7 PAC_CLIENT_INFO"
+    seq:
+      - id: client_id
+        -orig-id: ClientId
+        type: filetime
+        doc: A FILETIME structure in little-endian format that contains the Kerberos initial ticket-granting ticket (TGT) authentication time
+      - id: name_length
+        -orig-id: NameLength
+        type: u2
+        doc: An unsigned 16-bit integer in little-endian format that specifies the length, in bytes, of the Name field
+      - id: name
+        -orig-id: Name
+        type: str
+        size: name_length
+        encoding: UTF-16LE
+        doc: An array of 16-bit Unicode characters in little-endian format that contains the client's account name
+  upn_dns_info:
+    doc: "See [MS-PAC] 2.10 UPN_DNS_INFO"
+    seq:
+      - id: upn_length
+        -orig-id: UpnLength
+        type: u2
+        doc: An unsigned 16-bit integer in little-endian format that specifies the length, in bytes, of the UPN information
+      - id: upn_offset
+        -orig-id: UpnOffset
+        type: u2
+        doc: An unsigned 16-bit integer in little-endian format that contains the offset to the beginning of the UPN information, in bytes, from the beginning of the UPN_DNS_INFO structure
+      - id: dns_domain_name_length
+        -orig-id: DnsDomainNameLength
+        type: u2
+        doc: An unsigned 16-bit integer in little-endian format that specifies the length, in bytes, of the DNS information
+      - id: dns_domain_name_offset
+        -orig-id: DnsDomainNameOffset
+        type: u2
+        doc: An unsigned 16-bit integer in little-endian format that contains the offset to the beginning of the DNS information, in bytes, from the beginning of the UPN_DNS_INFO structure
+      - id: padding1
+        size: upn_offset - 8
+      - id: upn
+        type: str
+        size: upn_length
+        encoding: UTF-16LE
+      - id: padding2
+        size: dns_domain_name_offset - upn_offset - upn_length
+      - id: dns_domain_name
+        type: str
+        size: dns_domain_name_length
+        encoding: UTF-16LE
```

Comment:
I don't see a reason why not use `instances` - it's easier, less hacky and more descriptive:

```suggestion
    instances:
      upn:
        pos: upn_offset
        type: str
        size: upn_length
        encoding: UTF-16LE
      dns_domain_name:
        pos: dns_domain_name_offset
        type: str
        size: dns_domain_name_length
        encoding: UTF-16LE
```